packages/playwright-core/src/server/browserContext.ts

```diff
@@ -612,67 +610,53 @@ export abstract class BrowserContext extends SdkObject {
     return result;
   }
 
-  async _resetStorage(progress: Progress) {
-    const oldOrigins = this._origins;
-    const newOrigins = new Map(this._options.storageState?.origins?.map(p => [p.origin, p]) || []);
-    if (!oldOrigins.size && !newOrigins.size)
-      return;
-    let page = this.pages()[0];
-
-    // Do not mark this page as internal, because we will leave it for later reuse
-    // as a user-visible page.
-    page = page || await this.newPage(progress, false);
-    const interceptor = (route: network.Route) => {
-      route.fulfill({ body: '<html></html>' }).catch(() => {});
-    };
-    await page.addRequestInterceptor(progress, interceptor, 'prepend');
-
-    try {
-      for (const origin of new Set([...oldOrigins, ...newOrigins.keys()])) {
-        const frame = page.mainFrame();
-        await frame.gotoImpl(progress, origin, {});
-        await progress.race(frame.resetStorageForCurrentOriginBestEffort(newOrigins.get(origin)));
-      }
-
-      this._origins = new Set([...newOrigins.keys()]);
-      // It is safe to not restore the URL to about:blank since we are doing it in Page::resetForReuse.
-    } finally {
-      await page.removeRequestInterceptor(interceptor);
-    }
-  }
-
   isSettingStorageState(): boolean {
     return this._settingStorageState;
   }
 
-  async setStorageState(progress: Progress, state: NonNullable<channels.BrowserNewContextParams['storageState']>) {
+  async setStorageState(progress: Progress, state: channels.BrowserNewContextParams['storageState'], useExistingPage: boolean) {
```

Comment:
Make it clear that it does not set the state, but rather adds it for now. Or pass the bool!